### PR TITLE
librustc_mir => 2018

### PR DESCRIPTION
**Pull request by taiki-e**
Transitions `librustc_mir` to Rust 2018; cc #58099

TODO: ~~elided_lifetimes_in_paths~~

r? @Centril

### CODE REVIEW

File: src/librustc_mir/lib.rs [hunk 1]
```diff
@@ -29,28 +28,19 @@ Rust MIR: a lowered representation of Rust. Also: an experiment!
...
-extern crate syntax_pos;
-extern crate rustc_target;
-extern crate log_settings;
-extern crate rustc_apfloat;
-extern crate byteorder;
-extern crate core;
-extern crate smallvec;
 
 // Once we can use edition 2018 in the compiler,
 // replace this with real try blocks.
```
Comment (scottmcm) [inline src/librustc_mir/lib.rs]: I'm excited to use real `try` blocks now that you've done this; if you'd like to remove this macro as part of this change that'd be awesome.  (Definitely doesn't need to be part of this, though, so feel free not to do it if you'd rather.)

File: src/librustc_mir/lib.rs [hunk 2]
```diff
@@ -4,7 +4,6 @@ Rust MIR: a lowered representation of Rust. Also: an experiment!
 
 */
 
-#![feature(nll)]
```
Comment (matthewjasper) [inline src/librustc_mir/lib.rs]: Can this be kept, it gives better error messages in some cases and ensures that we can't ignore future compatibility warnings.

**Review by matthewjasper — CHANGES_REQUESTED** [review]
One nit and then I think that this is good to go.

File: src/librustc_mir/borrow_check/nll/mod.rs [hunk 3]
```diff
@@ -254,14 +254,14 @@ fn dump_mir_results<'a, 'gcx, 'tcx>(
     );
 
     // Also dump the inference graph constraints as a graphviz file.
-    let _: io::Result<()> = try_block! {
+    let _: io::Result<()> = try {
         let mut file =
             pretty::create_dump_file(infcx.tcx, "regioncx.all.dot", None, "nll", &0, source)?;
         regioncx.dump_graphviz_raw_constraints(&mut file)?;
     };
```
Comment (scottmcm) [inline src/librustc_mir/borrow_check/nll/mod.rs]: :tada:

File: src/librustc_mir/transform/qualify_consts.rs [hunk 4]
```diff
@@ -27,10 +27,10 @@ use syntax_pos::{Span, DUMMY_SP};
 use std::fmt;
 use std::usize;
 
-use transform::{MirPass, MirSource};
+use crate::transform::{MirPass, MirSource};
 use super::promote_consts::{self, Candidate, TempState};
 
-bitflags! {
+bitflags::bitflags! {
```
Comment (taiki-e) [inline src/librustc_mir/transform/qualify_consts.rs]: `bitflags!` is only used here.

**Review by matthewjasper — APPROVED** [review]

**Pull request merged**